broker.go

```diff
@@ -82,6 +83,18 @@ func (b *Broker) Open(conf *Config) error {
 		b.conn = newBufConn(b.conn)
 
 		b.conf = conf
+
+		if conf.Net.SASL.Enable {
+			err = b.doSASLPlainAuth()
+			if err != nil {
+				Logger.Printf("SASL authentication with broker %s failed\n", b.addr)
+				_ = b.Close()
```

Comment:
go's mutexes are not re-entrant, so this will deadlock; to handle the error please use `connErr` and follow the pattern established a few lines up when we fail to connect at all; note you will still need to close `b.conn` since it has now been established